flocker/node/agents/blockdevice.py

```diff
@@ -871,10 +1003,11 @@ def calculate_necessary_state_changes(self, local_state,
             for manifestation
             in manifestations_to_create
         )
+        # XXX: Maybe list volumes here and pass the list to each helper below.
+        deletes = self._calculate_deletes(configured_manifestations) # volumes=volumes)
```

Comment:
I'm struggling to see how this is going to work for resizes. I left a few comments in the updated design.